code/game/objects/items/weapons/RCD.dm

```diff
@@ -331,6 +339,32 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON)
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		var/datum/rcd_act/remove_user/act = new()
+		if(act.try_act(suicide_tile, src, user))
+			user.visible_message("<span class='suicide'>[user] deconstructs [user.p_themselves()] with [src]!</span>")
+			for(var/obj/item/W in user)	// Do not delete all their stuff.
+				user.unEquip(W)			// Dump everything on the floor instead.
+			flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+			user.dust()
+			return OBLITERATION
+		flags &= ~NODROP // If we got past the above check without dying, remove the superglue from the RCD.
+		return SHAME // Ensure that we do not accidentally perform the other flavour of suicide act.
```

Comment:
Could you also make this into an early return?

Reply:
It is done.